libvcs4j-spoon/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/SpoonModel.java

```diff
@@ -0,0 +1,336 @@
+                    .collect(Collectors.toList());
+
+            filesNotCompiledSinceLastUpdate.addAll(filesToBuild);
+
+            //delete the removed files from the spoon model
+            removeChangedTypes(getAllFilesFromFileChanges(revisionRange.getRemovedFiles(), true));
+
+            //delete removed files from the set, because they do not exist anymore
+            getAllFilesFromFileChanges(revisionRange.getRemovedFiles(), true)
+                    .forEach(path -> filesNotCompiledSinceLastUpdate.remove(path));
+
+            //delete relocated files from the set, because their path is outdated
+            getAllFilesFromFileChanges(revisionRange.getRelocatedFiles(), true)
+                    .forEach(path -> filesNotCompiledSinceLastUpdate.remove(path));
+
+            //remove the changed classes from spoon model
+            removeChangedTypes(filesNotCompiledSinceLastUpdate);
+
+            filesNotCompiledSinceLastUpdate.addAll(getAllFilesFromFileChanges(revisionRange.getAddedFiles(), false));
+
+            launcher.addInputResource(createInputSource(filesNotCompiledSinceLastUpdate));
+            launcher.getModelBuilder().addCompilationUnitFilter(path -> !filesToBuild.contains(path));
+            launcher.getModelBuilder().compile(SpoonModelBuilder.InputType.FILES);
+            model.get().setBuildModelIsFinished(false);
+            try {
+                model = Optional.of(launcher.buildModel());
+            } catch (Exception e) {
+                model = Optional.empty();
+                filesNotCompiledSinceLastUpdate.clear();
+            }
+
+        } else {
+
+            final Launcher launcher = new Launcher();
+            launcher.setBinaryOutputDirectory(temporaryDirectory);
+            launcher.getEnvironment().setNoClasspath(noClasspath);
+            //Add the checked out directory here, so we do not have to add each single file
+            launcher.addInputResource(revisionRange.getRevision().getOutput().toString());
+            launcher.getModelBuilder().compile(SpoonModelBuilder.InputType.FILES);
+            model = Optional.of(launcher.buildModel());
+
+        }
+        LOGGER.info(format("Model built in %d milliseconds", currentTimeMillis() - current));
+    }
+
+    /**
+     * Transforms the given Input into a {@link FilteringFolder}. Each ath in the given collection has to exist,
+     * be readable and be a regular Java source file, otherwise it will not be added to the {@link FilteringFolder}.
+     *
+     * @param input The given collection with absolute paths to source files.
+     * @return A {@link FilteringFolder} containing all files from the given input.
+     */
+    private FilteringFolder createInputSource(final Collection<String> input) {
+        FilteringFolder folder = new FilteringFolder();
+        input.forEach(path -> {
+            final Path p = Paths.get(path);
+            if (Files.exists(p) && Files.isReadable(p) && Files.isRegularFile(p)) {
```

Comment:
I think the file must also be writable if you want to delete it.
```suggestion
            if (Files.exists(p) && Files.isWritable(p) && Files.isRegularFile(p)) {
```